Sources/NetworkService/Request.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// Encapsulates an HTTP request with the necessary components.
+public struct Request {
+    public typealias HTTPHeaderKey = String
+    public typealias HTTPHeaderValue = String
+    public typealias HTTPHeaders = [HTTPHeaderKey: HTTPHeaderValue]
```

Comment:
Do we need to add the doc comments for the public interface?

Reply:
Fixed